packages/workspace/src/workspace/nacl_files/nacl_files_source.ts

```diff
@@ -247,17 +247,58 @@ export const toParsedNaclFile = async (
   }
 }
 
-const parseNaclFile = async (
-  naclFile: NaclFile, functions: Functions,
-): Promise<Required<ParseResult>> =>
-  (parse(Buffer.from(naclFile.buffer), naclFile.filename, functions))
+async function parseNaclFile(
+  {
+    naclFile,
+    functions,
+    createSourceMap,
+  }:
+  {
+    naclFile: NaclFile
+    functions: Functions
+    createSourceMap?: false
+  }
+): Promise<ParseResult>
+async function parseNaclFile(
+  {
+    naclFile,
+    functions,
+    createSourceMap,
+  }:
+  {
+    naclFile: NaclFile
+    functions: Functions
+    createSourceMap: boolean
+  }
+): Promise<Required<ParseResult>>
+
+async function parseNaclFile(
+  {
+    naclFile,
+    functions,
+    createSourceMap = false,
+  }:
+  {
+    naclFile: NaclFile
+    functions: Functions
+    createSourceMap?: boolean
+  }
+): Promise<ParseResult> {
+  return parse(Buffer.from(naclFile.buffer), naclFile.filename, functions, createSourceMap)
+}
+
+type ParseNaclFilesArgs = {
+  naclFiles: NaclFile[]
+  functions: Functions
+  createSourceMap?: boolean
+}
```

Comment:
```suggestion
type ParseNaclFilesArgs = {
  naclFiles: NaclFile[]
  functions: Functions
  createSourceMap?: boolean
}
async function parseNaclFile(args: ParseNaclFilesArgs & { createSourceMap?: false }): Promise<ParseResult>
async function parseNaclFile(args: ParseNaclFilesArgs & { createSourceMap: true }): Promise<Required<ParseResult>>
async function parseNaclFile(
  { naclFile, functions, createSourceMap = false }: ParseNaclFilesArgs
): Promise<ParseResult> {
  return parse(Buffer.from(naclFile.buffer), naclFile.filename, functions, createSourceMap)
}
```
RYL (not quite DRY, but repeat yourself less?)

Reply:
Unless I'm missing something, there's a difference between `ParseNaclFilesArgs` and `ParseNaclFileArgs`, which is that the `naclFiles` member is an array for `ParseNaclFilesArgs` but a single file for `ParseNaclFileArgs`. Let me see if I can take the spirit of this suggested change and build on it.